src/diffusers/pipelines/unclip/pipeline_unclip.py

```diff
@@ -0,0 +1,503 @@
+# TODO License
+
+from typing import Callable, List, Optional, Union
+
+import torch
+from torch import nn
+from torch.nn import functional as F
+
+import torchvision.transforms.functional as TVF
+from diffusers import PriorTransformer, UNet2DConditionModel, UNet2DModel
+from diffusers.modeling_utils import ModelMixin
+from diffusers.pipeline_utils import DiffusionPipeline, ImagePipelineOutput
+from diffusers.schedulers import DDPMScheduler
+from torchvision.transforms import InterpolationMode
+from transformers import CLIPTextModelWithProjection, CLIPTokenizer
+
+from ...configuration_utils import ConfigMixin, register_to_config
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class UnCLIPCLIPStats(ModelMixin, ConfigMixin):
+    @register_to_config
+    def __init__(self, clip_embeddings_dim: int = 768):
+        super().__init__()
+
+        self.clip_mean = nn.Parameter(torch.zeros(1, clip_embeddings_dim))
+        self.clip_std = nn.Parameter(torch.zeros(1, clip_embeddings_dim))
+
+    def forward(self, prior_latents):
+        prior_latents = (prior_latents * self.clip_std) + self.clip_mean
+        return prior_latents
+
+
+class UnCLIPEmbeddingUtils(ModelMixin, ConfigMixin):
```

Comment:
- a) Let's maybe call this `UnCLIPToTextProjModel` (or something that sounds like a neural network) 
- b) Let's move this model to its own file and import it from there (similar to how it's done for https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/audio_diffusion/mel.py)

Reply:
done!